tests/system/feature_store/test_feature_store.py

```diff
@@ -968,6 +968,210 @@ def test_filtering_parquet_by_time(self, with_tz, local):
         with pytest.raises(FileNotFoundError):
             result_offline_target.as_df()
 
+    @TestMLRunSystem.skip_test_if_env_not_configured
+    @pytest.mark.parametrize(
+        "partition_keys",
+        [
+            ["year"],
+            ["year", "month"],
+            ["year", "month", "day"],
+            ["year", "month", "day", "hour"],
+        ],
+    )
+    @pytest.mark.parametrize("with_tz", [True, False])
+    def test_partitioned_parquet_as_df(self, partition_keys, with_tz):
+        """
+        test reading partitioned parquet target as_df method with time filtering
+        covers:
+          - Partitioned parquet writing via ParquetTarget
+          - Reading & filtering via start_time/end_time
+          - Empty/out-of-range case
+        """
+        key = "patient_id"
+        base_time = datetime(2020, 12, 1, 17, 0)
+        if with_tz:
+            base_time = base_time.replace(tzinfo=pytz.UTC)
+
+        df = pd.DataFrame(
+            [
+                {
+                    key: i + 1,
+                    "timestamp": base_time + timedelta(hours=i),
+                    "value": i * 10,
+                }
+                for i in range(4)
+            ]
+        )
+
+        if partition_keys == ["year"]:
+            granularity = "year"
+        elif partition_keys == ["year", "month"]:
+            granularity = "month"
+        elif partition_keys == ["year", "month", "day"]:
+            granularity = "day"
+        elif partition_keys == ["year", "month", "day", "hour"]:
+            granularity = "hour"
+        else:
+            raise ValueError("Unexpected partition_keys")
```

Comment:
This list comes from us, so it should suffice to just make it

```suggestion
        granularity = partition_keys[-1]
```